src/Chainweb/CutDB.hs

```diff
@@ -455,10 +457,12 @@ startCutDb config logfun headerStore payloadStore cutHashesStore = mask_ $ do
                 Right hm -> unsafeMkCut v
                     <$> case _cutDbParamsInitialHeightLimit config of
                         Nothing -> return hm
-                        Just h -> limitCutHeaders wbhdb h hm
-                            -- FIXME: this requires that the pact db is deleted because it
-                            -- interfers with rewind limit. As a fix temporarily disable
-                            -- rewind limit during initialization.
+                        Just h -> do
+                            let
+                                cutHeightTarget =
+                                    avgCutHeightAt v h -
+                                        CutHeight (int $ diameterAtCutHeight v (maxBound :: CutHeight) * chainCountAt v (maxBound :: BlockHeight))
```

Comment:
Why is this term needed? 

Can `cutHeightTarget` become negative?

Reply:
This term compensates for the `avgCutHeightAt v h` potentially having some chains which are past the block height in question. In other words, with this term in here, we make sure that all of the chains are actually at or before the target block height at `cutHeightTarget`. Technically `cutHeightTarget` can become negative because of this but for the current chain graph we'd need to reset to a block height lower than 3.

Reply:
Let's add `min 0 ...` to make sure that we don't go past genesis?